### ISSUE TITLE
我把视频流用其他方法转化成h264 然后传给onSoftEncodedData方法   现象是黑屏  有声音  麻烦问下是什么原因呀

### ISSUE THREAD

**Issue by wangyongkai** [begeekmyfriend/yasea#746]
public void onSoftEncodedData(byte[] es, long pts, boolean isKeyFrame) {
        ByteBuffer bb = ByteBuffer.wrap(es);
        MediaCodec.BufferInfo vebi = new MediaCodec.BufferInfo();
        vebi.offset = 0;
        vebi.size = es.length;
        vebi.presentationTimeUs = pts;
        vebi.flags = isKeyFrame ? MediaCodec.BUFFER_FLAG_KEY_FRAME : 0;
        onEncodedAnnexbFrame(bb, vebi);
    }

    // when got encoded h264 es stream.
    private void onEncodedAnnexbFrame(ByteBuffer es, MediaCodec.BufferInfo bi) {
        mp4Muxer.writeSampleData(videoMp4Track, es.duplicate(), bi);
        flvMuxer.writeSampleData(videoFlvTrack, es, bi);
    }

**Comment by begeekmyfriend** [comment]
Knowledge is more important than code, please look up relevant materials through search engine and then checkout this project back again.

**Comment by wangyongkai** [comment]
You are absolutely right .   But right now I need to prioritize project functionality .   Finish first and study later

**Comment by begeekmyfriend** [comment]
https://github.com/begeekmyfriend/yasea/issues/291
https://github.com/begeekmyfriend/yasea/issues/594
Maybe you need custom `SrsEncoder.java` to meet your demand.

**Comment by wangyongkai** [comment]
视频源如果是编码后的h264  传给onSoftEncodedData方法后是黑屏   请大神给指点下啊  真的百思不解 十分感谢

**Comment by begeekmyfriend** [comment]
Please check whether the fields in input arguments `MediaCodec.BufferInfo` and `ByteBuffer` to be feeded into [onEncodedAnnexbFrame](https://github.com/begeekmyfriend/yasea/blob/master/library/src/main/java/net/ossrs/yasea/SrsEncoder.java#L344) are correct or not. And you might add some log in [writeVideoSample](https://github.com/begeekmyfriend/yasea/blob/master/library/src/main/java/net/ossrs/yasea/SrsFlvMuxer.java#L876) to see whether there is data flowing down.
